src/lib/string.ts

```diff
@@ -45,3 +45,109 @@ export function uncapitalize(value?: string): string | undefined {
 
   return value.charAt(0).toLowerCase() + value.slice(1);
 }
+
+/**
+ * Truncates a string to a maximum length, adding a suffix if truncated
+ * @param value The string to truncate
+ * @param maxLength The maximum length of the resulting string
+ * @param suffix The suffix to append if truncated (default: "...")
+ * @returns The truncated string
+ */
+export function truncate(value: string | null | undefined, maxLength: number, suffix = "..."): string {
+  if (!value || typeof value !== "string") {
+    return value || "";
+  }
+
+  if (value.length <= maxLength) {
+    return value;
+  }
+
+  return `${value.slice(0, maxLength)}${suffix}`;
+}
+
+/**
+ * Removes all occurrences of needle from the beginning and end of haystack
+ * @param haystack The string to trim
+ * @param needle The string to remove (default: " ")
+ * @returns The trimmed string
+ */
+export function trim(haystack: string | null | undefined, needle = " "): string {
+  if (!haystack || typeof haystack !== "string") {
+    return haystack || "";
+  }

```

Comment:
the check should be like in the other functions
```suggestion
  if (!value || isNullOrWhitespace(value)) {
    return value;
  }
```